// simple-weather-api/lib/weather.js
'use strict';
	
var apiKey;

/* Weather API endpoint */
var ENDPOINT = "http://api.openweathermap.org/data/2.5/weather?q=<KEYWORD>&APPID=<APP-KEY>";

/**
 * Module dependencies
 */
var http 	= require('http');


/**
 * Constructs a new class that provides access to Weather API
 * @constructor WeatherAPI
 * @param string API key
 */
var WeatherAPI = function(apiKeyString) {
    apiKey = apiKeyString;
};
	
/**
 * Getting the weather details based on given keyword
 * @keyword
 * @apiKey
 * @return
 */
WeatherAPI.prototype.getWeather = function(keyword)
{
	return new Promise(function(resolve, reject){
		var uri 	= ENDPOINT.replace("<KEYWORD>", keyword);
			uri		= uri.replace("<APP-KEY>", apiKey);
		
		http.get(uri, function(res) {
			//console.log("Got response: " + res.statusCode);
			
			var dataChunks = [];
			res.setEncoding('utf8');
			res.on("data", function(chunk) {
				dataChunks.push(chunk);
			});
			
			res.on("end", function() {
				var response = {
					headers: res.headers,
					body: dataChunks.join(''),
					statusCode: res.statusCode,
					statusMessage: res.statusMessage
				};
				
				//console.log("Body: " + response);
				resolve(response);
			});
		}).on("error", function(e) {
			//console.log("Error: " + e.message);
			reject(e.message);
		});
	});
};

module.exports = WeatherAPI;
